app/scripts/modules/dcos/pipeline/stages/runJob/runJobStage.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+import _ from 'lodash';
+
+const angular = require('angular');
+
+import { DcosProviderSettings } from '../../../dcos.settings';
+
+module.exports = angular.module('spinnaker.dcos.pipeline.stage.runJobStage', [
+  require('dcos/job/general.component.js'),
+  //TODO Add back when scheduled jobs are supported better by Spinnaker
+  //require('dcos/job/schedule.component.js'),
+  require('dcos/job/labels.component.js')
+])
+  .config(function(pipelineConfigProvider) {
+    pipelineConfigProvider.registerStage({
+      provides: 'runJob',
+      cloudProvider: 'dcos',
+      templateUrl: require('./runJobStage.html'),
+      executionDetailsUrl: require('./runJobExecutionDetails.html'),
+      validators: [
+        { type: 'requiredField', fieldName: 'account' },
+        { type: 'requiredField', fieldName: 'general.id' }
+      ]
+    });
+  }).controller('dcosRunJobStageCtrl', function($scope, accountService, $q) {
+    let stage = $scope.stage;
+    this.stage = $scope.stage;
+
+    if (!_.has(stage, 'name')) {
+      _.set(stage, 'name', Date.now().toString());
+    }
+
+    stage.cloudProvider = 'dcos';
+    stage.application = $scope.application.name;
+
+    if (!stage.credentials && $scope.application.defaultCredentials.dcos) {
+      stage.credentials = $scope.application.defaultCredentials.dcos;
+    }
+
+    this.accountChanged = () => {
+      setRegistry();
+      this.updateRegions();
+    };
+
+    this.regionChanged = () => {
+    };
+
+    this.updateRegions = () => {
+      if (stage.account) {
+        $scope.stage.dcosClusters = $scope.backingData.credentialsKeyedByAccount[stage.account].dcosClusters;
+        if ($scope.stage.dcosClusters.map(r => r.name).every(r => r !== stage.region)) {
+          this.regionChanged();
+          delete stage.region;
+        }
+      } else {
+        $scope.stage.dcosClusters = null;
+      }
+    };
+
+    this.onChange = (changes) => {
+      stage.docker.image.registry = changes.registry;
+    };
+
+    function attemptToSetValidAccount(accountsByName, stage) {
+      var defaultAccount = DcosProviderSettings.defaults.account;
+      var dcosAccountNames = _.keys(accountsByName);
+      var firstDcosAccount = null;
+      if (dcosAccountNames.length) {
+        firstDcosAccount = dcosAccountNames[0];
+      }
+
+      var defaultAccountIsValid = defaultAccount && dcosAccountNames.includes(defaultAccount);
+
+      stage.account =
+        defaultAccountIsValid ? defaultAccount : (firstDcosAccount ? firstDcosAccount : 'my-dcos-account');
```

Comment:
I may be misunderstanding the comment but I believe we're already doing that.  If `defaultAccountIsValid` is `false` then we use `firstDcosAccount` (which is `dcosAccountNames[0]`) unless there are no `dcosAccountNames` in which case we return the hardcoded `my-dcos-account` because we don't have anything else at that point.   We had seen something similar in a few of the other providers where they would fall back to a hardcoded value like `my-account-name`.  Would it be preferable to leave account null if we don't have any other account at that point?